### PR TITLE
Added default branch to a repository

### PR DESCRIPTION
**Pull request by oriash93**
will be used to select the default branch when creating Github pull request

### CODE REVIEW

File: Git.hub/Repository.cs [hunk 1]
```diff
@@ -87,10 +86,28 @@ public IList<Branch> GetBranches()
...
+        /// Retrieves the name of the default branch
+        /// </summary>
+        /// <returns>The name of the default branch</returns>
+        public string GetDefaultBranch()
+        {
+            RestRequest request = new RestRequest("/repos/{user}/{repo}");
+            request.AddUrlSegment("user", Owner.Login);
+            request.AddUrlSegment("repo", Name);
+
+            var repo = _client.Get<Repository>(request).Data;
```
Comment (RussKie) [inline Git.hub/Repository.cs:99]: Should this called be wrapped in `try...catch`? If it fails, would it crash the app?

File: Git.hub/Repository.cs [hunk 2]
```diff
@@ -87,10 +86,28 @@ public IList<Branch> GetBranches()
...
+
+            if (repo == null)
```
Comment (RussKie) [inline Git.hub/Repository.cs]: Please add braces

Comment (oriash93) [reply Git.hub/Repository.cs]: Same comment here.

File: Git.hub/Repository.cs (hunk 1 above)
Comment (oriash93) [reply Git.hub/Repository.cs:99]: To be honest, I followed the standard of the other methods in this class. Do you think I should change it?

File: Git.hub/Repository.cs (hunk 2 above)
Comment (RussKie) [reply Git.hub/Repository.cs]: Yes, please

File: Git.hub/Repository.cs (hunk 1 above)
Comment (RussKie) [reply Git.hub/Repository.cs:99]: Try throwing an exception from here and see where it detrimentally affects the hosting app.
If it does, we definitely need `try...catch`. If not, we can leave it as is.

**Comment by oriash93** [comment]
@RussKie  I decided to change the implementation, take a look.

And for your question:
> Try throwing an exception from here and see where it detrimentally affects the hosting app.
If it does, we definitely need try...catch. If not, we can leave it as is.

I tried causing it to crash (i.e. by going offline) but as I suspected, another method - `GetBranches()` threw a `NullReferenceException`.

**Comment by RussKie** [comment]
> I tried causing it to crash (i.e. by going offline) but as I suspected, another method - `GetBranches()` threw a `NullReferenceException`.

Has it caused the whole app to crash?

**Comment by oriash93** [comment]
> > I tried causing it to crash (i.e. by going offline) but as I suspected, another method - `GetBranches()` threw a `NullReferenceException`.
> 
> Has it caused the whole app to crash?

No. Maybe the exception is caught somewhere else, becuase It showed this window:
image

"Continue" resumes the form in the application. "Quit" closes the application.

The full exception details are:

```
System.NullReferenceException: Object reference not set to an instance of an object.
   at GitHub3.GitHubRepo.GetBranches() in D:\Repositories\gitextensions\Plugins\GitHub3\GitHubRepo.cs:line 78
   at GitUI.CommandsDialogs.RepoHosting.CreatePullRequestForm.<>c__DisplayClass22_0.<<PopulateBranchesComboAndEnableCreateButton>b__0>d.MoveNext() in D:\Repositories\gitextensions\GitUI\CommandsDialogs\RepoHosting\CreatePullRequestForm.cs:line 137
--- End of stack trace from previous location where exception was thrown ---
   at System.Runtime.ExceptionServices.ExceptionDispatchInfo.Throw()
   at System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task task)
   at System.Runtime.CompilerServices.ConfiguredTaskAwaitable.ConfiguredTaskAwaiter.GetResult()
   at GitUI.ThreadHelper.<>c__DisplayClass13_0.<<FileAndForget>b__0>d.MoveNext() in D:\Repositories\gitextensions\GitExtUtils\GitUI\ThreadHelper.cs:line 101
```

**Comment by RussKie** [comment]
I suspected as much, it crashes the app, and we need `try..catch` around these calls. Either here or in the app.

**Pull request merged**